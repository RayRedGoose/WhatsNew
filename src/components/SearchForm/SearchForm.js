import React, { Component } from 'react';
import './SearchForm.scss';
import PropTypes from 'prop-types';

class SearchForm extends Component {
  constructor(props) {
    super();
    this.state = {
      searchCondition: ''
    }
  }

  handleChange = event => {
    this.setState({ searchCondition: event.target.value });
  }

  keyPressed = event => {
    if (event.key === "Enter") {
      this.submitSearch(event)
    }
  }

  submitSearch = event => {
    event.preventDefault();
    const condition = this.state.searchCondition.toLowerCase();
    const searchResult = this.props.articleShown.filter(item => {
      return (
        item.headline.toLowerCase().includes(condition) ||
        item.description.toLowerCase().includes(condition)
      )
    })
    if (searchResult.length > 0) {
      this.props.changeArticleSwown(searchResult);
    }
    this.clearInputs();
  }

  clearInputs = () => {
    this.setState({ searchCondition: '' });
  }

  render() {
    return (
      <form className="search">
        <input
          type="text"
          placeholder="Search for news article here."
          onChange={this.handleChange}
          value={this.state.searchCondition}
          onKeyPress={this.keyPressed}/>
        <button onClick={this.submitSearch}>Search Now</button>
      </form>
    );
  }
}

SearchForm.propTypes = {
  changeArticleSwown: PropTypes.func,
  articleShown: PropTypes.array
}

export default SearchForm;
