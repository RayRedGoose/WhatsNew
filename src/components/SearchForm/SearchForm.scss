.search {
  grid-area: search;
  display: flex;
  justify-content: space-between;
  padding: 20px;
  * {
    font-size: 18px;
  }
  input {
    border: none;
    background: none;
    color: #fff;
    padding: 0.4em 10px;
    width: 80%;
  }
  button {
    background-color: #b66af0;
    border: none;
    border-radius: 50px;
    color: #fff;
    padding: 0.4em 2em;
    &:hover {
      transform: scale(1.1);
    }
    &:active {
      transform: scale(0.9);
    }
  }
}
