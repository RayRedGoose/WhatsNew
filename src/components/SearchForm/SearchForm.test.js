import React from 'react';
import { shallow } from 'enzyme';
import SearchForm from './SearchForm';

describe('Search Form', () => {
  let container,
      changeArticleSwownMock = jest.fn(),
      article = {
        id: "1",
        headline: 'Mysterious comet',
        img: './image.jpg',
        description: 'Mysterious comet will cause a rare \'Unicorn\' meteor storm this week.',
        url: 'https://www.example.com'
      };

  beforeEach(() => {
    container = shallow(<SearchForm
      changeArticleSwown={changeArticleSwownMock}
      articleShown={[ article ]}
    />);
  })

  it('should match the snapshot with all data passed in correctly', () => {
    expect(container).toMatchSnapshot();
  });

  it('should call the changeArticleSwown prop when clicked', () => {
    container.find('button').simulate(
      'click',
      {preventDefault: () => {}}
    );

    expect(changeArticleSwownMock).toHaveBeenCalledWith([ article ]);
  });

  it('should update state with a search condition when handleChange is called', () => {

    expect(container.state('searchCondition')).toEqual('');

    container.find('input').simulate(
      'change',
      {target: {value: 'matched'}}
    );

    expect(container.state('searchCondition')).toEqual('matched');
  });

  it('should update state with a search condition when clearInputs is called', () => {

    container.find('input').simulate(
      'change',
      {target: {value: 'tratata'}}
    );

    expect(container.state('searchCondition')).toEqual('tratata');

    container.find('button').simulate(
      'click',
      {preventDefault: () => {}}
    );

    expect(container.state('searchCondition')).toEqual('');
  });

  it('should call submitSearch when click was activated by enter', () => {
    const instance = container.instance();
    const spy = jest.spyOn(instance, "submitSearch").mockImplementation(() => {});

    container.find('input').simulate(
      'keypress',
      {key: 'Enter'}
    );

    expect(spy).toHaveBeenCalled();
  });
})
