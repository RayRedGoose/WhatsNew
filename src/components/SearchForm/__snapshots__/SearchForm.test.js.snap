// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search Form should match the snapshot with all data passed in correctly 1`] = `
<form
  className="search"
>
  <input
    onChange={[Function]}
    onKeyPress={[Function]}
    placeholder="Search for news article here."
    type="text"
    value=""
  />
  <button
    onClick={[Function]}
  >
    Search Now
  </button>
</form>
`;
